pysph/sph/bc/inlet_outlet_manager.py

```diff
@@ -0,0 +1,318 @@
+        x = []
+        y = []
+        z = []
+        all_info = self.inletinfo + self.outletinfo
+        for info in all_info:
+            x.append(info.bound[0][0])
+            x.append(info.bound[1][0])
+            y.append(info.bound[0][1])
+            y.append(info.bound[1][1])
+            z.append(info.bound[0][2])
+            z.append(info.bound[1][2])
+        xmax, xmin = max(x), min(x)
+        ymax, ymin = max(y), min(y)
+        zmax, zmin = max(z), min(z)
+
+        for info in all_info:
+            info.domain = [[xmin, ymin, zmin], [xmax, ymax, zmax]]
+
+    def add_io_properties(self, pa, scheme=None):
+        """Add properties to be used in inlet/outlet equations
+
+        Parameters
+        ----------
+
+        pa : particle_array
+            Particle array of inlet/outlet
+        scheme : pysph.sph.scheme
+            The insance of scheme class
+        """
+        pass
+
+    def get_io_names(self):
+        """return all the names of inlets and outlets"""
+        return self.inlets + self.outlets
+
+    def get_stepper(self, scheme, integrator, **kw):
+        """Returns the steppers for inlet/outlet
+
+        Parameters
+        ----------
+
+        scheme : pysph.sph.scheme
+            The instance of the scheme class
+        intergrator : pysph.sph.integrator
+            The parent class of the integrator
+        **kw : extra arguments
+            Extra arguments depending upon the scheme used
+        """
+        raise NotImplementedError()
+
+    def setup_iom(self, dim, kernel):
+        """Essential data passed
+
+        Parameters
+        ----------
+
+        dim : int
+            dimension of the problem
+        kernel : pysph.base.kernel
+            the kernel instance
+        """
+        self.dim = dim
+        self.kernel = kernel
+
+    def get_equations(self, scheme, **kw):
+        """Returns the equations for inlet/outlet
+
+        Parameters
+        ----------
+
+        scheme : pysph.sph.scheme
+            The instance of the scheme class
+        **kw : extra arguments
+            Extra arguments depending upon the scheme used
+        """
+        raise NotImplementedError()
+
+    def get_equations_post_compute_acceleration(self, scheme, **kw):
+        """Returns the equation for inlet/outlet post accelration
+           computation is done
+
+        Parameters
+        -----------
+
+        scheme : pysph.sph.scheme
+            The instance of the scheme class
+        **kw : extra arguments
+            Extra arguments depending upon the scheme used
+        ---------
+        """
+        pass
+
+    def get_inlet_outlet(self, particle_array):
+        """ Returns list of `Inlet` and `Outlet` instances which
+            performs the change in inlet particles to outlet
+            particles.
+
+        Parameters
+        -----------
+
+        particle_array : list
+            List of all particle_arrays
+        """
+        obj = []
```

Comment:
Use `result = []` or something more descriptive instead of `obj`.

Reply:
done